src/zope/app/form/browser/widgets.rst

```diff
@@ -13,56 +13,60 @@ description
 Simple stuff.
 
 Our Poll product holds an editable list of the `PollOption` instances.
-This is shown in the ``poll.py`` source below::
 
-    from persistent import Persistent
-    from interfaces import IPoll, IPollOption
-    from zope.interface import implements, classImplements
+First the Schemas are defined in the ``interfaces.py`` file below::
+
+    >>> from zope.interface import Interface
+    >>> from zope.schema import Object, Tuple, TextLine
+    >>> from zope.schema.interfaces import ITextLine
+    >>> from zope.i18nmessageid import MessageFactory
+
+    >>> _ = MessageFactory("poll")
+
+    >>> class IPollOption(Interface):
+    ...    label = TextLine(title=u'Label', min_length=1)
+    ...    description = TextLine(title=u'Description', min_length=1)
+
+    >>> class IPoll(Interface):
+    ...    options = Tuple(title=u'Options',
+    ...        value_type=Object(IPollOption, title=u'Poll Option'))
+    ...
+    ...    def getResponse(option): "get the response for an option"
+    ...
+    ...    def choose(option): 'user chooses an option'
+
+The implementation is shown in the ``poll.py`` source below::
+
+    >>> from persistent import Persistent
+    >>> from zope.interface import implementer, classImplements
+
+    >>> @implementer(IPollOption)
+    ... class PollOption(Persistent, object):
+    ...       pass
+
+    >>> @implementer(IPoll)
+    ... class Poll(Persistent, object):
+    ...
+    ...
+    ...  def getResponse(self, option):
```

Comment:
No way, vim does that? Out of the box or with custom plugins/configuration? 

I always figured it was a hard problem. But if vim can do it, emacs must be able to as well...I have some research to do.